test/lib/users/ssh_authorized_keyring_test.rb

```diff
@@ -167,7 +161,16 @@
         end
       end
 
-      context "when ssh directory already exists" do
+      context "when SSH directory is a symbolic link" do
+        it "raises a SSHDirectoryIsLink" do
```

Comment:
I'd make the assertion more strong with adding `... and does not write the SSH key` + add expectation `#save` to not be called.